kraken-build/src/kraken/std/python/buildsystem/uv.py

```diff
@@ -294,25 +278,49 @@ def get_lockfile(self) -> Path | None:
 
 
 class UvManagedEnvironment(ManagedEnvironment):
-    def __init__(self, project_directory: Path, uv_bin: str) -> None:
+    def __init__(self, project_directory: Path) -> None:
         self.project_directory = project_directory
-        self.uv_bin = uv_bin
-        self.env_path = project_directory / ".venv"
+        self._env_path: Path | None | NotSet = NotSet.Value
+
+    def _get_uv_environment_path(self) -> Path | None:
+        """Uses `uv run` to determines the location of the venv."""
+
+        # Ensure we de-activate any environment that might be active when Kraken is invoked. Otherwise,
+        # Poetry would fall back to that environment.
+        environ = os.environ.copy()
+        venv = get_current_venv(environ)
+        if venv:
+            venv.deactivate(environ)
+
+        command = ["uv", "run", "bash", "echo \$VIRTUAL_ENV"]
```

Comment:
This might be preferable to avoid the need for escaping:
```suggestion
        command = ["uv", "run", "bash", "printenv", "VIRTUAL_ENV"]
```

Reply:
This produces an error on my machine:
```
/usr/bin/printenv: /usr/bin/printenv: cannot execute binary file
```

Reply:
I can't reproduce on macOS or Ubuntu; but it's not a blocker, and we can retain your original approach.
```
$ uv run printenv USER
scott
```

Reply:
Actually, `printenv` works but I forgot the `-c` argument when calling `bash`.